### ISSUE TITLE
ListJSON failed to load config for crypt remote

### ISSUE THREAD

**Issue by egatenby** [rclone/rclone#7833]
#### The associated forum post URL from `https://forum.rclone.org`
https://forum.rclone.org/t/listjson-failed-to-load-config-for-crypt-remote/45915/1

#### What is the problem you are having with rclone?
When overriding any of the crypt: remote configuration on the command line, lsjson will fail to work if --encrypted is passed.  For example, passing  --crypt-strict-names or --crypt-password or --crypt-password2. This error apparently happens because the --crypt* command line options cause `detected overridden config - adding "{-7R5S}" suffix to name`. Later validations on this temporary config name cause this error:
```
ListJSON failed to load config for crypt remote: config name contains invalid characters - may only contain numbers, letters, _, -, ., +, @ and space, while not start with - or space, and not end with space
```

#### What is your rclone version (output from `rclone version`)
```
rclone v1.66.0
- os/version: Microsoft Windows 11 Pro 23H2 (64 bit)
- os/kernel: 10.0.22631.3447 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.22.1
- go/linking: static
- go/tags: cmount
```

and latest beta

```
rclone v1.67.0-beta.7927.b2f6aac75
- os/version: Microsoft Windows 11 Pro 23H2 (64 bit)
- os/kernel: 10.0.22631.3447 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.22.2
- go/linking: static
- go/tags: cmount
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Microsoft Windows 11 Pro 23H2 (64 bit)

#### Which cloud storage system are you using? (e.g. Google Drive)
crypt: wrapping a local:

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
```
.\rclone --config local.conf --crypt-password=x --crypt-password2=y -vv lsjson "Local_Crypted_V:scratch" --encrypted
```

My configuration:
```
[Local_Drive]
type = local

[Local_Crypted_V]
type = crypt
remote = Local_Drive:V:/rc/crypted
filename_encoding = base32768
password=x
password2=y
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
```
2024/05/07 23:19:25 DEBUG : rclone: Version "v1.66.0" starting with parameters ["C:\\backups\\rclone.exe" "--config" "local.conf" "--crypt-password=x" "--crypt-password2=y" "-vv" "lsjson" "Local_Crypted_V:scratch" "--encrypted"]
2024/05/07 23:19:25 DEBUG : Creating backend with remote "Local_Crypted_V:scratch"
2024/05/07 23:19:25 DEBUG : Using config file from "C:\\backups\\local.conf"
2024/05/07 23:19:25 DEBUG : Local_Crypted_V: detected overridden config - adding "{-7R5S}" suffix to name
2024/05/07 23:19:26 DEBUG : Creating backend with remote "Local_Drive:V:/rc/crypted/倡㓵熽䥠籋ጰ赝♽㪿"
2024/05/07 23:19:26 DEBUG : fs cache: renaming cache item "Local_Drive:V:/rc/crypted/倡㓵熽䥠籋ጰ赝♽㪿" to be canonical "Local_Drive://?/V:/rc/crypted/倡㓵熽䥠籋ጰ赝♽㪿"
2024/05/07 23:19:26 DEBUG : fs cache: switching user supplied name "Local_Drive:V:/rc/crypted/倡㓵熽䥠籋ጰ赝♽㪿" for canonical name "Local_Drive://?/V:/rc/crypted/倡㓵熽䥠籋ጰ赝♽㪿"
2024/05/07 23:19:26 DEBUG : fs cache: renaming cache item "Local_Crypted_V:scratch" to be canonical "Local_Crypted_V{-7R5S}:scratch"
[
2024/05/07 23:19:26 DEBUG : 2 go routines active
2024/05/07 23:19:26 Failed to lsjson: ListJSON failed to load config for crypt remote: config name contains invalid characters - may only contain numbers, letters, `_`, `-`, `.`, `+`, `@` and space, while not start with `-` or space, and not end with space
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
Thanks for reporting this.

This should be easy to fix - please give this a try

[v1.67.0-beta.7930.175b8364e.fix-7833-lsjson-crypt](https://beta.rclone.org/branch/fix-7833-lsjson-crypt/v1.67.0-beta.7930.175b8364e.fix-7833-lsjson-crypt/) on branch [fix-7833-lsjson-crypt](https://github.com/rclone/rclone/tree/fix-7833-lsjson-crypt) (uploaded in 15-30 mins)

**Comment by egatenby** [comment]
[v1.67.0-beta.7930.175b8364e.fix-7833-lsjson-crypt](https://beta.rclone.org/branch/fix-7833-lsjson-crypt/v1.67.0-beta.7930.175b8364e.fix-7833-lsjson-crypt/) worked.  thank you!

**Comment by ncw** [comment]
Thank you for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.67
